### ISSUE TITLE
Tasker:Task crashes when choosing the plugin from within Tasker

### ISSUE THREAD

**Issue by yochananmarqos** [termux/termux-tasker#3]
I created the /data/data/com.termux/files/home/.termux/tasker folder and placed a shell script in it. The crash happens whether there is anything in that folder or not.

Nexus 5X
Android 7.1.1

Logcat:
```
---- Feb 12, 2017 11:20:17 AM ----

02-12 11:15:48.823  2234 28829 I ActivityManager: START u0 {act=com.twofortyfouram.locale.intent.action.EDIT_SETTING cmp=com.termux.tasker/.EditConfigurationActivity (has extras)} from uid 10130 on display 0

02-12 11:15:48.929  2234  2253 I ActivityManager: Start proc 1062:com.termux.tasker/u0a239 for activity com.termux.tasker/.EditConfigurationActivity

02-12 11:15:49.180  1062  1062 E AndroidRuntime: Process: com.termux.tasker, PID: 1062

02-12 11:15:49.180  1062  1062 E AndroidRuntime: java.lang.RuntimeException: Unable to start activity ComponentInfo{com.termux.tasker/com.termux.tasker.EditConfigurationActivity}: java.lang.NullPointerException: Attempt to get length of null array

02-12 11:15:49.180  1062  1062 E AndroidRuntime: 	at com.termux.tasker.EditConfigurationActivity.onCreate(Unknown Source)

02-12 11:15:49.183  2234  9747 W ActivityManager:   Force finishing activity com.termux.tasker/.EditConfigurationActivity

02-12 11:15:49.696  2234  2266 W ActivityManager: Activity pause timeout for ActivityRecord{fb4f60e u0 com.termux.tasker/.EditConfigurationActivity t2776 f}

02-12 11:15:50.195  2234  5122 I ActivityManager: Process com.termux.tasker (pid 1062) has died

02-12 11:15:50.198  2234  9747 W ActivityManager: Ignoring remove of inactive process: ProcessRecord{762dfc4 0:com.termux.tasker/u0a239}

02-12 11:16:06.399  2234  6107 I ActivityManager: START u0 {act=com.twofortyfouram.locale.intent.action.EDIT_SETTING cmp=com.termux.tasker/.EditConfigurationActivity (has extras)} from uid 10130 on display 0

02-12 11:16:06.420  2234 25764 I ActivityManager: Start proc 1511:com.termux.tasker/u0a239 for activity com.termux.tasker/.EditConfigurationActivity

02-12 11:16:06.625  1511  1511 E AndroidRuntime: Process: com.termux.tasker, PID: 1511

02-12 11:16:06.625  1511  1511 E AndroidRuntime: java.lang.RuntimeException: Unable to start activity ComponentInfo{com.termux.tasker/com.termux.tasker.EditConfigurationActivity}: java.lang.NullPointerException: Attempt to get length of null array

02-12 11:16:06.625  1511  1511 E AndroidRuntime: 	at com.termux.tasker.EditConfigurationActivity.onCreate(Unknown Source)

02-12 11:16:06.630  2234  2252 W ActivityManager:   Force finishing activity com.termux.tasker/.EditConfigurationActivity

02-12 11:16:07.135  2234  2266 W ActivityManager: Activity pause timeout for ActivityRecord{5f84d0c u0 com.termux.tasker/.EditConfigurationActivity t2776 f}

02-12 11:16:07.594  2234 22499 I ActivityManager: Process com.termux.tasker (pid 1511) has died

02-12 11:16:07.598  2234  2252 W ActivityManager: Ignoring remove of inactive process: ProcessRecord{bb7711b 0:com.termux.tasker/u0a239}

02-12 11:16:19.717  2234  6111 I ActivityManager: START u0 {act=com.twofortyfouram.locale.intent.action.EDIT_SETTING cmp=com.termux.tasker/.EditConfigurationActivity (has extras)} from uid 10130 on display 0

02-12 11:16:19.753  2234 28829 I ActivityManager: Start proc 1742:com.termux.tasker/u0a239 for activity com.termux.tasker/.EditConfigurationActivity

02-12 11:16:19.916  1742  1742 E AndroidRuntime: Process: com.termux.tasker, PID: 1742

02-12 11:16:19.916  1742  1742 E AndroidRuntime: java.lang.RuntimeException: Unable to start activity ComponentInfo{com.termux.tasker/com.termux.tasker.EditConfigurationActivity}: java.lang.NullPointerException: Attempt to get length of null array

02-12 11:16:19.916  1742  1742 E AndroidRuntime: 	at com.termux.tasker.EditConfigurationActivity.onCreate(Unknown Source)

02-12 11:16:19.918  2234  5122 W ActivityManager:   Force finishing activity com.termux.tasker/.EditConfigurationActivity

02-12 11:16:20.426  2234  2266 W ActivityManager: Activity pause timeout for ActivityRecord{a012eaf u0 com.termux.tasker/.EditConfigurationActivity t2776 f}

02-12 11:16:20.723  2234  9749 I ActivityManager: Process com.termux.tasker (pid 1742) has died

02-12 11:18:27.966  2234  4662 I ActivityManager: START u0 {act=com.twofortyfouram.locale.intent.action.EDIT_SETTING cmp=com.termux.tasker/.EditConfigurationActivity (has extras)} from uid 10130 on display 0

02-12 11:18:28.074  2234 25765 I ActivityManager: Start proc 5345:com.termux.tasker/u0a239 for activity com.termux.tasker/.EditConfigurationActivity

02-12 11:18:28.159  5345  5345 W System  : ClassLoader referenced unknown path: /data/app/com.termux.tasker-1/lib/arm64

02-12 11:18:28.342  5345  5345 E AndroidRuntime: Process: com.termux.tasker, PID: 5345

02-12 11:18:28.342  5345  5345 E AndroidRuntime: java.lang.RuntimeException: Unable to start activity ComponentInfo{com.termux.tasker/com.termux.tasker.EditConfigurationActivity}: java.lang.NullPointerException: Attempt to get length of null array

02-12 11:18:28.342  5345  5345 E AndroidRuntime: 	at com.termux.tasker.EditConfigurationActivity.onCreate(Unknown Source)

02-12 11:18:28.344  2234  2253 W ActivityManager:   Force finishing activity com.termux.tasker/.EditConfigurationActivity

02-12 11:18:28.852  2234  2266 W ActivityManager: Activity pause timeout for ActivityRecord{dfe94 u0 com.termux.tasker/.EditConfigurationActivity t2776 f}

02-12 11:18:32.027  2234 28830 I ActivityManager: Process com.termux.tasker (pid 5345) has died

02-12 11:18:32.031  2234  2253 W ActivityManager: Ignoring remove of inactive process: ProcessRecord{2126603 0:com.termux.tasker/u0a239}

---- Feb 12, 2017 11:20:17 AM ----
```

**Comment by fornwall** [comment]
Does these steps reproduce your crash, or do you do anything different?

1. Start the Tasker app.
2. Select the `Tasks` tab at the top.
3. Press the `+` button at the bottom of the screen.
4. Write a task name and press the checkbox to create it.
5. Press the `+` button at the bottom of the screen.
6. In the `Selection Action Category` dialog shown, select `Plugin`.
7. In the `Action Plugin` dialog shown, select `Termux:Task`.
8. It crashes?

**Comment by yochananmarqos** [comment]
Almost. It crashes when I select the Configuration icon after choosing the Termux:Task Plugin.

**Comment by fornwall** [comment]
@yochananmarqos Just after the crash, can you run `logcat -d > log.txt` in Termux and then upload the resulting log.txt file somewhere and link to it?

**Comment by yochananmarqos** [comment]
@fornwall Here you go: https://www.dropbox.com/s/zmwb69jb4mnz5wu/log.txt?dl=0

**Comment by agnostic-apollo** [comment]
This was likely due to a call to `TASKER_DIR.listFiles().length`, but should be solved in version [0.5](https://github.com/termux/termux-tasker/releases/tag/v0.5) since there is now a [null check](https://github.com/termux/termux-tasker/blob/3e70898d7921e3634e4a45f259356253ecff1ce0/app/src/main/java/com/termux/tasker/EditConfigurationActivity.java#L371).
